src/api/app/controllers/cloud/upload_jobs_controller.rb

```diff
@@ -0,0 +1,47 @@
+module Cloud
+  class UploadJobsController < ApplicationController
+    include UploadJobHelper
+
+    before_action :require_login
+    before_action -> { feature_active?(:cloud_upload) }
+    before_action :validate_configuration_presence, only: [:index, :create]
+
+    def index
+      render xml: ::Cloud::Backend::UploadJob.all(::User.current, format: :xml)
+    end
+
+    def show
+      if ::User.current.upload_jobs.where(job_id: params[:id]).exists?
+        render xml: ::Cloud::Backend::UploadJob.find(params[:id], format: :xml)
+      else
+        render_error status: 404
+      end
+    end
+
+    def create
+      upload_job = ::Cloud::UploadJob.create(upload_data.merge(user: ::User.current))
+      if upload_job.valid?
+        render xml: ::Cloud::Backend::UploadJob.find(upload_job.user_upload_job.id, format: :xml)
+      else
+        render_error status: 400,
+                     errorcode: 'cloud_upload_job_invalid',
+                     message: "Failed to create upload job: #{upload_job.errors.full_messages.to_sentence}."
+      end
+    end
+
+    private
+
+    def validate_configuration_presence
+      return if ::User.current.cloud_configurations?
+      render_error status: 400,
```

Comment:
[nitpick] Would ``:bad_request`` work here?
(as you used a symbol in feature_active and I also think that rubocop will complain with the new update)

Reply:
That's a test helper

Reply:
Hm, how does it work then in application_controller?